x-pack/plugin/ent-search/qa/rest/src/yamlRestTest/resources/rest-api-spec/test/entsearch/90_behavioral_analytics_event_post.yml

```diff
@@ -0,0 +1,163 @@
+setup:
+  - do:
+      behavioral_analytics.put:
+        name: my-test-analytics-collection
+
+---
+teardown:
+  - do:
+      behavioral_analytics.delete:
+        name: my-test-analytics-collection
+        ignore: 404
+
+---
+"Post Pageview Analytics Event":
+  - do:
+      behavioral_analytics.post_event:
+        collectionName: my-test-analytics-collection
+        eventType: "pageview"
+        body:
+            session:
+              id: "123"
+            user:
+              id: "456"
+            page:
+              url: "https://www.elastic.co"
+---
+"Post Analytics Pageview Event - missing page url":
+  - do:
+      catch: "bad_request"
+      behavioral_analytics.post_event:
+        collectionName: my-test-analytics-collection
+        eventType: "pageview"
+        body:
+          session:
+            id: "123"
+  - match: { error.type: "illegal_argument_exception" }
```

Comment:
we might match the error message to ensure it's the parameter we think (see [here](https://github.com/elastic/elasticsearch/blob/main/modules/reindex/src/yamlRestTest/resources/rest-api-spec/test/delete_by_query/20_validation.yml#L12) for example)

Reply:
Because of the hierachical nature of the parsing it is not that easy to test these messages in YAML REST tests.
I choose to test error messages pretty extensively in unit tests instead.